### PR TITLE
Add pgsql 16 testing

### PR DESCRIPTION
**Pull request by MaceWindu**
- support PostgreSQL 16 testing
- remove PostgreSQL 10 from CI (EOL)
- removed unused by CI `netcoreapp31` test configs
- updated some deps

### CODE REVIEW

File: Tests/Linq/Linq/FromSqlTests.cs [hunk 1]
```diff
@@ -503,7 +503,7 @@ static T AsTyped<T>(string str)
 
 		[Test]
 		public void TestInvaildAliasExprUsage(
-			[IncludeDataSources(TestProvName.AllPostgreSQL)]
+			[IncludeDataSources(TestProvName.AllPostgreSQL15Minus)]
```
Comment (MaceWindu) [inline Tests/Linq/Linq/FromSqlTests.cs:506]: This is the only significant change. Looked like PGSQL16 bug at first, but actually a new feature:

[Allow subqueries in the FROM clause to omit aliases](https://www.postgresql.org/docs/16/release-16.html#RELEASE-16-GENERAL)

File: Directory.Packages.props [hunk 2]
```diff
@@ -23,8 +23,8 @@
 		<PackageVersion Include="Nullability.Source"                                    Version="2.2.0"         />
 		<PackageVersion Include="PolySharp"                                             Version="1.13.2"        />
 		<!--build support-->
-		<PackageVersion Include="Microsoft.CodeAnalysis.BannedApiAnalyzers"             Version="3.11.0-beta1.23364.2" />
```
Comment (viceroypenguin) [inline Directory.Packages.props:26]: Microsoft.CodeAnalysis.BannedApiAnalyzers moved location in the file. should it be moved back to reduce churn?

**Review by viceroypenguin — APPROVED** [review]
lgtm

**Pull request merged**